emission/analysis/userinput/matcher.py

```diff
@@ -76,83 +77,60 @@ def handle_multi_non_deleted_match(confirmed_obj, ui):
         logging.warn("Invalid status found in user input %s, moving forward anyway" % ui)
 
 def create_confirmed_objects(user_id):
-    time_query = epq.get_time_range_for_confirmed_object_creation(user_id)
+    tq = epq.get_time_range_for_confirmed_object_creation(user_id)
     try:
-        # we will query the same time range for trips and places,
-        # but querying 'enter_ts' for places and 'end_ts' for trips
-        time_query.timeType = "data.enter_ts"
-        processed_places = create_confirmed_places(user_id, time_query)
-        time_query.timeType = "data.end_ts"
-        last_expected_trip_done = create_confirmed_trips(user_id, time_query, processed_places)
-        if last_expected_trip_done is None:
-            logging.debug("after run, last_expected_trip_done == None, must be early return")
-            epq.mark_confirmed_object_creation_done(user_id, None)
-        else:
-            epq.mark_confirmed_object_creation_done(user_id, last_expected_trip_done.data.end_ts)
-
+        ts = esta.TimeSeries.get_time_series(user_id)
+        timeline = esdtl.get_timeline(user_id,
+                           esda.CLEANED_PLACE_KEY,
+                           esda.EXPECTED_TRIP_KEY,
+                           esda.CLEANED_UNTRACKED_KEY,
+                           tq.startTs, tq.endTs, trip_id_key='cleaned_trip')
+        keys = eac.get_config()["userinput.keylist"]
+        
+        last_processed_ts = None
+        cTrip = cPlace = untracked = False
+        for tle in timeline:
+            logging.debug("timeline entry = %s" % tle)
+            if tle['metadata']['key'] == esda.CLEANED_PLACE_KEY:
+                cPlace = create_confirmed_entry(ts, tle, esda.CONFIRMED_PLACE_KEY, keys)
+            elif tle['metadata']['key'] == esda.EXPECTED_TRIP_KEY:
+                cTrip = create_confirmed_entry(ts, tle, esda.CONFIRMED_TRIP_KEY, keys)
+            elif tle['metadata']['key'] == esda.CLEANED_UNTRACKED_KEY:
+                untracked = True
+
+            if cPlace:
+                logging.debug("Creating confirmed place")
+                cp_id = ts.insert(cPlace)
+                if cTrip:
+                    logging.debug("Found trip that came before place, inserting confirmed trip with confirmed place id")
+                    cTrip['data']['confirmed_place'] = cp_id
+                    ts.insert(cTrip)
+                elif untracked:
+                    logging.debug("Found place that follows untracked time, confirmed trip will not be created")
```

Comment:
I don't understand this. `untracked_time` is intended to be `trip_like` with a start and end timestamp. So why do you log this as a place?
So the timeline around untracked time should be 

```
... -> cleaned place -> expected trip -> cleaned place -> untracked time -> cleaned place -> expected trip -> ...
```

If I understand this logic correctly, you will end up with 

```
... -> confirmed place -> confirmed trip -> confirmed place ->  -> confirmed place -> confirmed trip -> ...

```

Is that what you expect? I would like to see some unit tests around this code and the new real_examples. You can use `emission/tests/analysisTests/intakeTests/TestPipelineRealData.py` for an example of how to load data and run the pipeline.

You can check the results manually or automatically against a saved result.

Reply:
I will reword that log statement - I see how it is confusing.

>Is that what you expect?
Yes, because I thought we did not want to create a confirmed object for untracked time. When composite trips are created, we just use the cleaned_untracked object

I will make unit tests to verify this is happening